tabular/src/autogluon/tabular/models/lr/lr_rapids_model.py

```diff
@@ -51,10 +49,52 @@ def _set_default_params(self):
 
     def _preprocess(self, X, **kwargs):
         X = super()._preprocess(X=X, **kwargs)
-        if not isinstance(X, np.ndarray):
+        if hasattr(X, 'toarray'):  # Check if it's a sparse matrix
             X = X.toarray()
         return X
 
     def _fit(self, X, y, **kwargs):
-        kwargs.pop("sample_weight", None)  # sample_weight is not supported
-        super()._fit(X=X, y=y, **kwargs)
+        """
```

Comment:
Are we not using `num_gpus` argument to the `_fit` call? How do we tell cuML how many GPUs to use?